src/toggle-switch/toggle-switch.scss

```diff
@@ -0,0 +1,235 @@
+.ToggleSwitch {
+  align-items: center;
+  display: inline-flex;
+
+  &:hover {
+    .ToggleSwitch-knob {
+      background-color: var(--color-btn-hover-bg);
+    }
+  }
+
+  &:active {
+    .ToggleSwitch-knob {
+      background-color: var(--color-btn-active-bg);
+    }
+  }
+}
+
+.ToggleSwitch-track {
+  position: relative;
+  display: block;
+  width: 64px;
+  height: 32px;
+  padding: 0;
+  text-decoration: none;
+  cursor: pointer;
+  user-select: none;
+  background-color: var(--color-switch-track-bg);
+  border-color: var(--color-switch-track-border);
+  border-style: $border-style;
+  border-width: $border-width;
```

Comment:
Unfortunately stylelint doesn't like this:

```
src/toggle-switch/toggle-switch.scss
 174:3  ✖  Please use a border width variable instead of "$border-style". See                               primer/borders
           https://primer.style/css/utilities/borders.                                                                    
 174:3  ✖  Please use a border style variable instead of "$border-width". See                               primer/borders
           https://primer.style/css/utilities/borders.
```

Reply:
Turns out it should be

```scss
border: $border-width $border-style var(--color-switch-track-border);
```